### PR TITLE
Fix asan error reported in kernel_collector_test TearDown(); add more UT framework.

### PR DESCRIPTION
**Pull request by jimwsplk**

### CODE REVIEW

File: collector/kernel/kernel_collector_test.cc [hunk 1]
```diff
@@ -210,70 +240,117 @@ class KernelCollectorTest : public ::testing::Test {
...
       system(
-          "echo starting workload 1; exec 1> /tmp/workload1.log 2>&1; set -x; whoami; pwd; ls; cd /tmp; pwd; ls; cd /; pwd; ls; cd ~; pwd; ls; echo workload 1 complete");
+          "exec 1> /tmp/workload-sockets.log 2>&1; echo starting workload; /root/src/test/workload/sockets/sockets.py 10 20; echo workload complete");
     });
-    thr2_ = std::thread([]() {
+
+    start_workload([]() {
       system(
-          "echo starting workload 2; /root/src/test/workload/sockets/sockets.py 10 20 > /dev/null; echo workload 2 complete");
+          "exec 1> /tmp/workload-curl.log 2>&1; echo starting workload; for n in $(seq 1 10); do curl google.com; done; echo workload complete");
```
Comment (jesafar) [inline collector/kernel/kernel_collector_test.cc:269]: Does this mean this test will only pass if outbound networking and DNS is enabled for the pod running the test?

**Review by jesafar — APPROVED** [review]
Would have liked a bit more commentary-documentation.  Is the message and foreach in the TestChannel meant for replay ability?

File: collector/kernel/kernel_collector_test.cc (hunk 1 above)
Comment (jimwsplk) [reply collector/kernel/kernel_collector_test.cc:269]: Yes, but this test isn't really testing anything specific at the moment, just that a minimum number of messages of certain types were sent, and is meant more as a starting point for actual tests, a set of examples of things that can be tested, and a good way to investigate/learn how the kernel-collector works.

**Comment by jimwsplk** [comment]
> Would have liked a bit more commentary-documentation. Is the message and foreach in the TestChannel meant for replay ability?

Yes, documentation is needed, but baby steps :^)

The for_each could be used to analyze messages and/or to potentially replay/generate/send input to certain components (with some more work).

**Pull request merged**